### PR TITLE
Fix almost all PMD errors, except for the God class one

### PR DESCRIPTION
**Pull request by ghyl**
AnimationHandler gives false positive for too many methods for unknown reason.

<bug, testing

### CODE REVIEW

File: src/test/java/nl/tudelft/ti2206/semgr8/model/piece/BombGemTest.java [hunk 1]
```diff
@@ -76,12 +74,14 @@ public void testDoPowerUp() {
...
         BombGem bombGem = (BombGem)b.getGem(0, 2);
         bombGem.doPowerUp(b);
-        assertTrue(b.getPiece(0, 1).isEmpty());
-        assertTrue(b.getPiece(0, 2).isEmpty());
-        assertTrue(b.getPiece(0, 3).isEmpty());
-        assertTrue(b.getPiece(1, 1).isEmpty());
-        assertTrue(b.getPiece(1, 2).isEmpty());
-        assertTrue(b.getPiece(1, 3).isEmpty());
+
+        int[][] removedPiecesCoordinates = {
```
Comment (kiliangrashoff) [inline src/test/java/nl/tudelft/ti2206/semgr8/model/piece/BombGemTest.java]: Would be better to parametrize this

File: src/test/java/nl/tudelft/ti2206/semgr8/model/piece/LightningGemTest.java [hunk 2]
```diff
@@ -5,8 +5,8 @@
 import org.junit.Before;
 import org.junit.Test;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
+import static junit.framework.TestCase.assertEquals;
```
Comment (kiliangrashoff) [inline src/test/java/nl/tudelft/ti2206/semgr8/model/piece/LightningGemTest.java]: deprecated import, use org.junit.Assert instead

File: src/test/java/nl/tudelft/ti2206/semgr8/model/piece/LightningGemTest.java [hunk 3]
```diff
@@ -5,8 +5,8 @@
...
+import static junit.framework.TestCase.assertTrue;
```
Comment (kiliangrashoff) [inline src/test/java/nl/tudelft/ti2206/semgr8/model/piece/LightningGemTest.java]: same as above

File: src/test/java/nl/tudelft/ti2206/semgr8/model/piece/LightningGemTest.java [hunk 4]
```diff
@@ -49,19 +49,19 @@ public void testToString() {
...
-        assertTrue(b.getPiece(6, 0).isEmpty());
-        assertTrue(b.getPiece(6, 1).isEmpty());
-        assertTrue(b.getPiece(5, 2).isEmpty());
-        assertTrue(b.getPiece(5, 3).isEmpty());
-        assertTrue(b.getPiece(4, 4).isEmpty());
-        assertTrue(b.getPiece(4, 5).isEmpty());
-        assertTrue(b.getPiece(3, 6).isEmpty());
-        assertTrue(b.getPiece(3, 7).isEmpty());
+
+        int[][] removedPiecesCoordinates = {
```
Comment (kiliangrashoff) [inline src/test/java/nl/tudelft/ti2206/semgr8/model/piece/LightningGemTest.java]: Better as parametrized test

**Comment by kiliangrashoff** [comment]
Looks good, can be merged when comments are processed.

**Comment by ghyl** [comment]
Parameterized tests added and comments processed! Will merge when CI is finished

**Pull request merged**